### ISSUE TITLE
SCM Git import does not work with a proxy

### ISSUE THREAD

**Issue by robertdebock** [rundeck/rundeck#1777]
When configuring RUNDECK in an environment where a proxy is required, Git import with a URL to an https repository does not work.

The error message I receive is:

```
cannot open git-upload-pack; github.com
```

Maybe there is a way to set a proxy, but the documentation does not show this.

I'll be happy to assist in debugging and/or modifying documentation.

**Comment by cosmocracy** [comment]
I'm investigating this same issue.  Have you made any progress, @robertdebock?  I've been attempting to inject the various -D settings so that Java will see the proxy server.  Still tinkering so I'll update here if I make progress.  I'm surprised that fewer people have issues--I thought Rundeck would be at play in organizations with proxy servers...

**Comment by cosmocracy** [comment]
Ok, using the command ```docker exec -it rundeck bash``` I was able to explore a bit to find out what was (and wasn't) working related to my proxied access to the internet (and GitHub) from inside the container.  In particular when I tried to ```ssh cosmocracy@github.com``` I received useful errors that helped me isolate the layers of problem.  Your details will, of course, vary.

Here's what I found I needed to do to make it work:
* Use the ```--dns``` argument on my ```docker run``` to ensure name resolution worked inside my container.  Note that my organization apparently egress-filters DNS so I needed to use my actual internal DNS servers instead of the 8.8.8.8 server many use
* Pass ```http_proxy``` and ```https_proxy``` environment variables into the ```docker run```
* Pass a ```RDECK_JVM``` environment variable that specified the Java ```-D``` flag to trigger use of the system proxies.

Here's a _scrubbed_ version of my ```docker run``` in case it's helpful (replace 1.2.3.4 with your internal DNS, 2.3.4.5 with your proxy server's IP, myhostname with your Rundeck server's hostname, "*" with your passwords, etc.):
```bash
sudo docker run --dns 1.2.3.4 -p 4440:4440 -e RUNDECK_ADMIN_PASSWORD=* -e RUNDECK_PASSWORD=* -e SERVER_URL=http://myhostname:4440 -e RDECK_JVM="-Djava.net.useSystemProxies=true" -e http_proxy="http://2.3.4.5:8090" -e https_proxy="http://2.3.4.5:8090" --name rundeck -t jordan/rundeck:latest
```

Once launched with the setting above, I not longer get the generic ```cannot open git-upload-pack``` error but  instead get an error related to the SSH keys (since I haven't yet set them up).

Hope this helps and if so you might close this Rundeck issue. -EK
